pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -452,6 +452,15 @@ func updateConditions(config *deployapi.DeploymentConfig, newStatus *deployapi.D
 	// Condition about progress.
 	if latestRC != nil {
 		switch deployutil.DeploymentStatusFor(latestRC) {
+		case deployapi.DeploymentStatusNew:
+			// In case we fail to created the deployer pod within strategy
+			// timeoutSeconds do not leave the deployment config in New state forever
+			// but timeout.
+			if deployutil.ConfigExceededTimeoutSeconds(config, latestRC) {
```

Comment:
@mfojtik I think you need to cancel that deployment (RC) as well .

Reply:
i will rather transition it to fail.

Reply:
this is the DC part, cancellation of latest deployment must be part of the Handle() because I need client access.